src/main/scala/wdltool/GraphPrint.scala

```diff
@@ -1,71 +1,101 @@
 package wdltool
 
 import java.nio.file.{Files, Paths}
+import java.util.concurrent.atomic.AtomicInteger
+
+import wdl4s.wdl.{CallOutput, Declaration, If, Scatter, WdlCall, _}
+import wdl4s.wdl.WdlGraphNode
 
-import wdl4s.{CallOutput, Declaration, If, Scatter, _}
 import scala.collection.JavaConverters._
 
+import cats._
+import cats.implicits._
+
 object GraphPrint {
 
-  case class WorkflowDigraph(workflowName: String, digraph: Set[String])
+  case class WorkflowDigraph(workflowName: String, digraph: NodesAndLinks)
+  case class NodesAndLinks(nodes: Set[String], links: Set[String])
+  implicit val scopesAndLinksMonoid = new Monoid[NodesAndLinks] {
+    override def empty: NodesAndLinks = NodesAndLinks(Set.empty, Set.empty)
+    override def combine(x: NodesAndLinks, y: NodesAndLinks): NodesAndLinks = NodesAndLinks(x.nodes |+| y.nodes, x.links |+| y.links)
+  }
+
+  sealed trait Link{
+    def linkString: String
+  }
+
+  case class GraphLink(linkString: String, scopePath: List[Scope])
 
   def generateWorkflowDigraph(file: String, allNodesMode: Boolean): WorkflowDigraph = {
     // It's ok to use .get here, we're happy to throw an exception and crash the program!
     val namespace = WdlNamespaceWithWorkflow.load(Files.readAllLines(Paths.get(file)).asScala.mkString(System.lineSeparator()), Seq(WdlNamespace.fileResolver _)).get
 
-    val digraph = if (allNodesMode) {
-      listAllGraphNodes(namespace)
-    } else {
-      val executables = GraphPrint.listExecutableGraphNodes(namespace.workflow)
-      listAllGraphNodes(namespace, graphNode => executables.contains(graphNode))
-    }
+    val digraph = listAllGraphNodes(namespace.workflow)
 
     WorkflowDigraph(namespace.workflow.unqualifiedName, digraph)
   }
 
-  private def defaultFilter: GraphNode => Boolean = _ => true
+  private def defaultFilter: WdlGraphNode => Boolean = _ => true
+
+  val clusterCount: AtomicInteger = new AtomicInteger(0)
+
 
-  private def listAllGraphNodes(namespace: WdlNamespaceWithWorkflow, filter: GraphNode => Boolean = defaultFilter): Set[String] = {
+  private def listAllGraphNodes(namespace: Scope): NodesAndLinks = {
```

Comment:
I meant "namespace" in the colloquial sense, not an official wdl object. But really I was just giving you an easy rationalization to get out of changing things, so up to you :)